cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1374,22 +1384,91 @@ private KafkaVersionChange getKafkaVersionChange(StatefulSet kafkaSts) {
+                                        Annotations.ANNO_STRIMZI_IO_MANUAL_ZK_SCALE + " annotation to false.";
+
+                                log.info(scaleMessage);
+
+                                zkManualScalingCondition
+                                        .withReason("replicaChange")
+                                        .withStatus("True")
+                                        .withMessage(scaleMessage);
+
+                                // Setting this will add the annotation to the STS in the state.zkStatefulSet() method
+                                zkManualScaling = true;
+
+                            } else if (Annotations.booleanAnnotation(sts, Annotations.ANNO_STRIMZI_IO_MANUAL_ZK_SCALE, false)) {
+                                // This is an on-going manual scaling operation. Once the annotation has been set to true by the operator
+                                // only the user can set it to false.
+                                String scaleMessage = "Detected manual Zookeeper scaling operation in progress. " +
+                                        "Rolling of Zookeeper cluster will remain disabled until " +
+                                        Annotations.ANNO_STRIMZI_IO_MANUAL_ZK_SCALE + " annotation is set to false.";
 
+                                log.info(scaleMessage);
+
+                                zkManualScalingCondition
+                                        .withReason("manualScaleAnnotationSet")
+                                        .withStatus("True")
+                                        .withMessage(scaleMessage);
+
+                                zkManualScaling = true;
+
+                            } else {
+                                // Either this is a ZK 3.4.x cluster or the user is not manually scaling the 3.5.x cluster.
+                                zkManualScalingCondition
+                                        .withStatus("False")
+                                        .withReason("manualScalingNotRequired");
+
+                                if (KafkaVersion.compareDottedVersions(zkCluster.getVersion(), "3.4.99") > 0) {
+                                    zkManualScalingCondition
+                                            .withMessage("Manual scaling of Zookeeper cluster is finished or not required");
+                                } else {
+                                    zkManualScalingCondition
+                                            .withMessage("Manual scaling of Zookeeper cluster is not needed for Zookeeper 3.4.x clusters");
+                                }
+
+                                zkManualScaling = false;
+                            }
+
+                            KafkaStatus kStatus = this.kafkaStatus;
+                            kStatus.setConditions(Collections.singletonList(zkManualScalingCondition.build()));
+                            this.updateStatus(kStatus).setHandler(statusResult -> {
```

Comment:
Do you mean for the Future returned by this method to have completed before the status has been updated?

Reply:
Good point, I will compose it with the final returned future